File: gencode/java/udmi/schema/BuildingConfigTranslation.java

package udmi.schema;

import javax.annotation.processing.Generated;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.annotation.JsonPropertyOrder;


/**
 * Building Config Translation
 * <p>
 * [Discovery result](../docs/specs/discovery.md) with implicit enumeration
 * 
 */
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonPropertyOrder({
    "present_value",
    "units",
    "states"
})
@Generated("jsonschema2pojo")
public class BuildingConfigTranslation {

    /**
     * dotted path to present_value field
     * 
     */
    @JsonProperty("present_value")
    @JsonPropertyDescription("dotted path to present_value field")
    public String present_value;
    @JsonProperty("units")
    public Object units;
    @JsonProperty("states")
    public Object states;

    @Override
    public int hashCode() {
        int result = 1;
        result = ((result* 31)+((this.units == null)? 0 :this.units.hashCode()));
        result = ((result* 31)+((this.present_value == null)? 0 :this.present_value.hashCode()));
        result = ((result* 31)+((this.states == null)? 0 :this.states.hashCode()));
        return result;
    }

    @Override
    public boolean equals(Object other) {
        if (other == this) {
            return true;
        }
        if ((other instanceof BuildingConfigTranslation) == false) {
            return false;
        }
        BuildingConfigTranslation rhs = ((BuildingConfigTranslation) other);
        return ((((this.units == rhs.units)||((this.units!= null)&&this.units.equals(rhs.units)))&&((this.present_value == rhs.present_value)||((this.present_value!= null)&&this.present_value.equals(rhs.present_value))))&&((this.states == rhs.states)||((this.states!= null)&&this.states.equals(rhs.states))));
    }

}

File: gencode/python/udmi/schema/building_config_translation.py
"""Generated class for building_config_translation.json"""


class BuildingConfigTranslation:
  """Generated schema class"""

  def __init__(self):
    self.present_value = None
    self.units = None
    self.states = None

  @staticmethod
  def from_dict(source):
    if not source:
      return None
    result = BuildingConfigTranslation()
    result.present_value = source.get('present_value')
    result.units = source.get('units')
    result.states = source.get('states')
    return result

  @staticmethod
  def map_from(source):
    if not source:
      return None
    result = {}
    for key in source:
      result[key] = BuildingConfigTranslation.from_dict(source[key])
    return result

  @staticmethod
  def expand_dict(input):
    result = {}
    for property in input:
      result[property] = input[property].to_dict() if input[property] else {}
    return result

  def to_dict(self):
    result = {}
    if self.present_value:
      result['present_value'] = self.present_value # 5
    if self.units:
      result['units'] = self.units # 5
    if self.states:
      result['states'] = self.states # 5
    return result
